data_scrapers/san_francisco_county.py

```diff
@@ -4,101 +4,238 @@
-def get_tests() -> Dict:
-    """Get tests by day, order by date ascending"""
-    date_order_query = '?$order=result_date'
-    return get_json(tests_url + date_order_query)
-
-def get_test_totals() -> Dict:
-    """Get total tests and total number of positive tests, to date"""
-    total_test_query = '?$select=sum(tests) as tests'
-    tests = get_json(tests_url + total_test_query)
-    total_positives_query = '?$select=sum(pos) as total_positives'
-    positives = get_json(tests_url + total_positives_query)
-    tests[0].update(positives[0])
-    return tests
+    response = requests.get(race_ethnicity_url)
+    response.raise_for_status()
+    # Dict of target_label : source_label for re-keying.
+    # Note: Native_Amer not currently reported
+    RACE_ETH_KEYS = {"Latinx_or_Hispanic": "Hispanic or Latino", "Asian": "Asian", "African_Amer": "Black or African American",
+                 "White": "White", "Pacific_Islander": "Native Hawaiian or Other Pacific Islander", "Native_Amer": "Native American", "Multiple_Race": "Multiple Race",
+                 "Other": "Other", "Unknown": "Unknown"}
+    data = response.json()
+    # re-key and aggregate to flatten race x ethnicity
+    race_eth_data: Dict[str,int] = { k:0 for k in RACE_ETH_KEYS.keys() } # initalize all categories to 0 for aggregating
+
+    for item in data: # iterate through all race x ethnicity objects
+        cases = int(item["confirmed_cases"])
+        cols = item.keys()
+
+        # handle unknown
+        # sum over only items for which BOTH race and ethnicity are Unknown, or one is Unknown and the other is not reported
+        if "race" in cols and "ethnicity" in cols:
+            if item["race"] == "Unknown" and item["ethnicity"] == "Unknown": # both race and ethnicity are Unknown
+                race_eth_data["Unknown"] += cases
+        elif "ethnicity" in cols and item["ethnicity"] == "Unknown": # or, race not reported and ethnicity is unknown
+            race_eth_data["Unknown"] += cases
+        elif "race" in cols and item["race"] == "Unknown": # or, ethnicity not reported and race is unknown
+            race_eth_data["Unknown"] += cases
+        
+        # sum over 'Hispanic or Latino', all races
+        if "ethnicity" in cols and item["ethnicity"] == "Hispanic or Latino":
+            race_eth_data["Latinx_or_Hispanic"] += cases
+        
+        # sum over all known races
+        if "race" in cols and item["race"] != "Unknown": 
+            if item["race"] == "Other":  # except for race = Other, ethnicity = Hispanic or Latino; exclude Other/Hispanic Latino from Other
+                if "ethnicity" in cols and item["ethnicity"] != "Hispanic or Latino": 
+                    race_eth_data["Other"] += cases
+            else:
+                for k,v in RACE_ETH_KEYS.items(): # look up this item's re-key
+                    if v == item["race"]:
+                        race_eth_data[k] += cases
```

Comment:
Are we double-counting when `ethnicity == 'Hispanic or Latino'` here? I think `{race: 'Unknown', ethnicity: 'Hispanic or Latino'}` and `{race: 'White', ethnicity: 'Hispanic or Latino'}` are both in the data and would wind up getting here AND to line 225, but your screenshot from the dashboard shows totals that sum up to 100%.

Reply:
Yes we were double counting for White, Hispanic or Latino as @kwonangela7 [notes](https://github.com/sfbrigade/data-covid19-sfbayarea/pull/52#discussion_r424857992). For any other race and Hispanic or Latino, I believe the individual gets counted for both their race and as Hispanic Latino. I guess SF excludes White & Hispanic or Latino from White. I'll edit this so it lines up with SF.

Reply:
Ah, sorry, I totally missed that last paragraph in her comment!

I guess I was reacting to the column in your screenshot being labeled “percent of cases” and noticing that they add up to 100%. That implies we shouldn’t double-count like this, although I agree that seems unexpected — if you’re doing anything other than just showing each (race × ethnicity) combination, you *should* get > 100%. 😕

> I'll edit this so it lines up with SF.

I didn’t follow the conversations about this when creating the data model closely, so sorry if this is already well-trodden ground: given the confusion you’re encountering around reverse-engineering SF’s logic here, is matching even what we actually want? If so, 👍